src/controllers/searchController.js

```diff
@@ -0,0 +1,32 @@
+const asyncHandler = require("express-async-handler");
+const Location = require("../models/locationModel");
+
+
+// @desc Get search results
+// @route GET /api/search
+// FOR SEARCH /api/search?description=search - example /api/search?description=cool
+// FOR CATEGORY /api/search?category=category - example /api/search?category=nature
+// FOR HASHTAG /api/search?hashtag=#hashtag - example /api/search?hashtag=%23picture - %23 translates to #
+
+// @acces Public
+const textSearch = asyncHandler(async (req, res) => {
+    const query = req.query
+
+    if (req.query.description != null){
```

Comment:
When we try to access a nonexistent property, then it will evaluate into `undefined`, not `null`. The only reason for why this works is because you use the `!=` instead of `!==`, which is what you should usually use. As far as I understand, using `===` / `!==` is preferred as it avoids some type errors. I think it would be better if you rewrote this if as:
`if (req.query.description !== undefined)`.